src/compiler/rust.rs

```diff
@@ -540,15 +549,26 @@ impl<T> CompilerHasher<T> for RustHasher
             for h in source_hashes.into_iter().chain(extern_hashes) {
                 m.update(h.as_bytes());
             }
-            // 6. TODO: Environment variables: CARGO_* at the very least.
-            // https://github.com/mozilla/sccache/issues/89
+            // 6. Environment variables. Ideally we'd use anything referenced
+            // via env! in the program, but we don't have a way to determine that
+            // currently, and hashing all environment variables is too much, so
+            // we'll just hash the CARGO_ env vars and hope that's sufficient.
+            // Upstream Rust issue tracking getting information about env! usage:
+            // https://github.com/rust-lang/rust/issues/40364
+            for &(ref var, ref val) in env_vars.iter() {
```

Comment:
Oops, good catch! Might have even caught that with the unit test I added if I had used more than one env var...